### DISCUSSION TITLE
Dynamic Import returns LoadableComponent instead of Module

### DISCUSSION THREAD

**Discussion by renatodex** [vercel/next.js#13417] [Help] [answered]
# Bug report

## Describe the bug

When I try to load Phaser using `dynamic imports` in a JSX component, it fails to load the `Phaser` object.
Instead, the loaded object is a `LoadableComponent`, and when I try to create the Phaser object, I receive this error:

`TypeError: Phaser.Game is not a Constructor` (see screenshot below)

## To Reproduce

I've created a sample repository to make it easy for anyone to reproduce the error:
https://github.com/renatodex/react-phaser-issue

The repo has 2 a `components` folder with 2 files using different techniques to import `Phaser`.

### 1. components/this_module_works.js

This file is a simple JS module, and I've used an `await` import on the `phaser` module to retrieve it only in client side (i check for `typeof window !== 'undefined'`)
It's imported on the `index.js` page and it works normally.

### 2. components/this_component_is_broken.js

Here, the idea is to make it a Component to be used on the main page.
I tried to use the `dynamic import` to retrieve the `phaser` module, but it always returns a `LoadableComponent` instance, instead of returning the `Phaser` object.

As a result, the page throws an error when you try to import it.

## Expected behavior

The error happens because the Dynamic Import returns a `LoadableComponent`, and I was expecting it to return the `Phaser` object as a result.

The TypeError is basically thrown because `LoadableComponent` is not the `Phaser` object.

## Screenshots

![image](https://user-images.githubusercontent.com/68507/82956291-e08d1d80-9f86-11ea-8fbf-ca04254f81d0.png)

## System information

- macOS
- chrome
- Version of Next.js: 9.4.2
- Version of Node.js: v12.14.1
- Version of Phaser: 3.23

## Additional context

The repository is a simple Next app. You can run the server using `yarn run dev` or something.

**Comment by Timer** [comment, marked as answer]
`next/dynamic` is only for components, not any dynamic imports. You should be using pure `import()`!

**Reply by renatodex** [reply]
Hey @Timer !
That makes a lot of sense now!

However, I must say that the documentation confused me on that:

If you look at: https://nextjs.org/docs/advanced-features/dynamic-import
It states the following:
```
Next.js supports ES2020 dynamic import() for JavaScript. 
With it you can import JavaScript modules (inc. React Components) 
dynamically and work with them. They also work with SSR.
```

It's saying that you can import Javascript modules, **including React Components**.
Maybe we should change that to make a more explicit statement?

**Reply by Timer** [reply]
Please send a docs update!

**Reply by garrytrinder** [reply]
This just caught me out as well trying to import the `@azure/msal-browser` node module, however as @Timer suggests using `import()` resolved this for me.

```js
import('@azure/msal-browser').then((msal) => {
  const msalInstance = new msal.PublicClientApplication(msalConfig);
  console.log(msalInstance)
});
```

**Reply by breadadams** [reply]
> Please send a docs update!

@Timer PR opened here https://github.com/vercel/next.js/pull/16299

**Reply by fr-nevin** [reply]
By using import(), is it supposed to come out as a separate chunk? @Timer
